benchmarks/transform_warp.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+from skimage.transform import SimilarityTransform, warp
+from skimage.util.dtype import convert
+
+
+class WarpSuite:
+    params = ([np.uint8, np.uint16, np.float32, np.float64],
+              [128, 1024, 4096],
+              [0, 1, 3]
+              )
+    param_names = ['dtype_in', 'N', 'order']
+
+    def setup(self, dtype_in, N, order):
+        self.image = convert(np.random.rand(N, N), dtype=dtype_in)
+        self.tform = SimilarityTransform(scale=1, rotation=np.pi / 10,
+                                         translation=(0, 4))
+        self.order = order
+
+    def same_type(self):
```

Comment:
I think you'll need to  prefix these functions with [`time_`](https://asv.readthedocs.io/en/stable/writing_benchmarks.html#timing) otherwise asv won't run / find them. And the signature must be able to accept all parameters as positional arguments because asv passes them implicitly to all `setup`, `time_x` and `peakmem_` methods of the class.